jormungandr-watchdog/examples/stdin_echo.rs

```diff
@@ -83,20 +90,109 @@ impl Service for StdoutWriter {
+        LogLevel::WARN => tracing::Level::WARN,
+        LogLevel::DEBUG => tracing::Level::DEBUG,
+        LogLevel::ERROR => tracing::Level::ERROR,
+        LogLevel::TRACE => tracing::Level::TRACE,
+    };
+}
+
+impl Settings for LoggerConfig {
+    fn add_cli_args<'a, 'b>() -> Vec<Arg<'a, 'b>> {
+        vec![Arg::with_name("Log level")
+            .short("ll")
+            .long("log_level")
+            .takes_value(true)
+            .default_value("Warning")
+            .value_name("LOG_LEVEL")
+            .help("Services log level: []")]
+    }
+
+    fn matches_cli_args<'a>(&mut self, matches: &ArgMatches<'a>) {
+        if let Some(level) = matches.value_of("cfg") {
+            self.level = match level.to_lowercase().as_str() {
+                "info" => LogLevel::INFO,
+                "warn" => LogLevel::WARN,
+                "debug" => LogLevel::DEBUG,
+                "error" => LogLevel::ERROR,
+                "trace" => LogLevel::TRACE,
+                _ => self.level.clone(),
+            };
+        }
+    }
+}
+
+struct LoggerService {
+    state: ServiceState<Self>,
+}
+
+#[async_trait]
+impl Service for LoggerService {
+    const SERVICE_IDENTIFIER: &'static str = "logger";
+    type State = service::NoState;
+    type Settings = LoggerConfig;
+    type IntercomMsg = service::NoIntercom;
+
+    fn prepare(state: ServiceState<Self>) -> Self {
+        let subscriber = Subscriber::builder()
+            .with_max_level(tracing::Level::TRACE)
+            .finish();
+        tracing::subscriber::set_global_default(subscriber)
+            .expect("setting tracing default failed");
+        LoggerService { state }
+    }
+
+    async fn start(mut self) {
+        let this = &mut self;
+        while let Some(cfg) = this.state.settings().updated().await {
+            let subscriber = Subscriber::builder()
+                .with_max_level(log_level_into_tracing_level(&cfg.level))
+                .finish();
+            tracing::subscriber::set_global_default(subscriber)
+                .expect("setting tracing default failed");
+            print!("{:?}", cfg);
+        }
+    }
+}
+
 #[derive(CoreServices)]
 struct StdEcho {
     stdin: service::ServiceManager<StdinReader>,
     stdout: service::ServiceManager<StdoutWriter>,
+    logger: service::ServiceManager<LoggerService>,
 }
 
 fn main() {
-    use tracing_subscriber::{fmt, EnvFilter};
-
-    let subscriber = fmt::Subscriber::builder()
-        .with_env_filter(EnvFilter::from_default_env())
-        .finish();
-
-    tracing::subscriber::set_global_default(subscriber).expect("setting tracing default failed");
+    // let subscriber = fmt::Subscriber::builder()
+    //     .with_env_filter(EnvFilter::from_default_env())
+    //     .finish();
+    //
+    // tracing::subscriber::set_global_default(subscriber).expect("setting tracing default failed");
 
     let app = App::new("stdin_echo");
     let watchdog = WatchdogBuilder::<StdEcho>::new(app).build();
```

Comment:
Right now it compiles but it does not run properly. Failing with `'called `Result::unwrap()` on an `Err` value: Message("missing field `stdin`", Some(Pos { marker: Marker { index: 7, line: 1, col: 7 }, path: "." }))'`
I'm checking it.

Reply:
still failing here?

Reply:
Yes, we should open a ticket to check this. I believe is a missing file, although not sure why would it not drop to the default settings anyway.